File: Quiz-web/src/app/modules/core/models/announcement.ts
import {Imaged} from "./imaged";
import {Image} from "./image";

export class Announcement implements Imaged {
  id: number;
  title: string;
  subtitle: string;
  fullText: string;
  authorLogin: string;
  authorId: number;
  isPublished: boolean;
  createdDate: Date;
  imageId: number = -1;
  image: Image;
}

File: Quiz-server/src/main/java/com/qucat/quiz/repositories/entities/Announcement.java
package com.qucat.quiz.repositories.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.util.Date;

@Data
@Builder
@AllArgsConstructor
public class Announcement {
    private int id;
    private String authorLogin;
    private boolean isPublished;
    private Date createdDate;
    private String title;
    private String subtitle;
    private String fullText;
    private int imageId;
    private Image image;
}
